src/encoded/static/components/cart/cart.js

```diff
@@ -411,65 +683,164 @@ class FileFacets extends React.Component {
+                            return selectedTerms[filteredSelectedFacetKey].indexOf(filePropValue) !== -1;
+                        });
 
-        // Detect if we have empty facets.
-        let emptyFacets = false;
-        if (displayedFacets) {
-            emptyFacets = !Object.keys(displayedFacets).some(field => displayedFacets[field].terms && displayedFacets[field].terms.length > 0);
+                        // A match means to add to the count of the current facet field file term
+                        // only.
+                        if (match) {
+                            FileFacets.addFileTermToFacet(facets, selectedFacetField, file);
+                        }
+                    });
+                }
+            });
+
+            // We need to include selected terms that happen to have a zero count, so add all
+            // selected facet terms not yet included in `facets`.
+            Object.keys(selectedTerms).forEach((field) => {
+                if (field in facets) {
+                    // Find selected terms NOT in facets and add them with a zero count.
+                    const missingTerms = selectedTerms[field].filter(term => Object.keys(facets[field]).indexOf(term) === -1);
+                    if (missingTerms.length > 0) {
+                        missingTerms.forEach((term) => {
+                            facets[field][term] = 0;
+                        });
+                    }
+                }
+            });
         }
 
+        return { facets: Object.keys(facets).length > 0 ? facets : null, selectedFileCount };
+    }
+
+    /**
+     * Called when the user clicks a facet expander button. Updates the expander states so the
+     * facets re-render to the new expanded states.
+     * @param {string} field Field name for clicked facet expander
+     * @param {bool}   altKey True if alt/option key down at time of click
+     */
+    handleExpanderClick(field, altKey) {
+        this.setState((state) => {
+            if (altKey) {
+                // Alt key held down, so expand or collapse *all* facets.
+                const nextExpandedState = !state.expanded[field];
+                const expandedStates = {};
+                displayedFacetFields.forEach((facetField) => {
+                    expandedStates[facetField] = nextExpandedState;
+                });
+                return { expanded: expandedStates };
+            }
+
+            // Alt key not held down, so just expand or collapse the clicked facet.
+            const nextExpandedStates = Object.assign({}, state.expanded);
+            nextExpandedStates[field] = !nextExpandedStates[field];
+            return { expanded: nextExpandedStates };
+        });
+    }
+
+    render() {
+        const { selectedTerms, termClickHandler } = this.props;
+
+        // Based on carted files and currently selected facet terms, generate a facet object for
```

Comment:
I think you mean `cart` not `carted`?

Reply:
I mean “files that are included in the cart,” so “carted files.” Is that weird?

Reply:
I think 'cart files' could also mean files in the cart, but 'carted files' works!